src/ripple/app/tx/impl/CFTokenIssuanceCreate.cpp

```diff
@@ -0,0 +1,127 @@
+//------------------------------------------------------------------------------
+/*
+  This file is part of rippled: https://github.com/ripple/rippled
+  Copyright (c) 2023 Ripple Labs Inc.
+
+  Permission to use, copy, modify, and/or distribute this software for any
+  purpose  with  or without fee is hereby granted, provided that the above
+  copyright notice and this permission notice appear in all copies.
+
+  THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+  WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+  MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+  ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+  WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+  ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/app/tx/impl/CFTokenIssuanceCreate.h>
+#include <ripple/ledger/View.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/protocol/TxFlags.h>
+#include <ripple/protocol/st.h>
+
+namespace ripple {
+
+NotTEC
+CFTokenIssuanceCreate::preflight(PreflightContext const& ctx)
+{
+    if (!ctx.rules.enabled(featureCFTokensV1))
+        return temDISABLED;
+
+    if (auto const ret = preflight1(ctx); !isTesSuccess(ret))
+        return ret;
+
+    if (ctx.tx.getFlags() & tfCFTokenIssuanceCreateMask)
+        return temINVALID_FLAG;
+
+    if (auto const fee = ctx.tx[~sfTransferFee])
+    {
+        if (fee > maxTransferFee)
+            return temBAD_CFTOKEN_TRANSFER_FEE;
+
+        // If a non-zero TransferFee is set then the tfTransferable flag
+        // must also be set.
+        if (fee > 0u && !ctx.tx.isFlag(tfTransferable))
+            return temMALFORMED;
+    }
+
+    // TODO length?
+    // if (auto const code = ctx.tx[~sfAssetCode]; code.length() >
+    // maxAssetCodeLength)
+    //     return temMALFORMED;
```

Comment:
Not necessarily disagreeing with your comment -- I like the simplicity. However, one rationale for choosing 160 bits here was to, as much as possible, keep parity with the [existing currency code](https://xrpl.org/currency-formats.html#currency-codes) format so that tooling like SDKs, etc wouldn't have to handle another scheme.

Reply:
I don't disagree with that rationale either.  But, if we were to change the way we represent currency codes now would be the time.  And there are motivations for such a change: both space savings and easing human perception by requiring names of a certain human readable format.

I'm not insisting that this change is worthwhile.  But we'd be remiss if we didn't recognize the opportunity being presented and think about the choice.  Once we've thought it through it may make sense the reject it.

Reply:
Note for posterity: I started a currency code discussion over here: https://github.com/XRPLF/XRPL-Standards/discussions/128

Reply:
we have agreed not to change the currency code format